shell/browser/ui/x/x_window_utils.cc

```diff
@@ -86,9 +83,11 @@ void MoveWindowAbove(x11::Window window, x11::Window other_window) {
 }
 
 bool IsWindowValid(x11::Window window) {
-  XWindowAttributes attrs;
-  return XGetWindowAttributes(gfx::GetXDisplay(), static_cast<uint32_t>(window),
-                              &attrs);
+  auto* conn = x11::Connection::Get();
+  auto attributes_req = conn->GetWindowAttributes({window});
+  conn->Flush();
```

Comment:
Is this `Flush` call required ?

Does `auto attributes_req = conn->GetWindowAttributes({window}).Sync();` not work ?

Reply:
@deepak1556 i based this off of the change made in the linked CL - they called Flush so it seemed to me we should as well

Reply:
we don't need it here, `.Sync()` will block until a response is received from the server https://source.chromium.org/chromium/chromium/src/+/master:ui/gfx/x/xproto_types.h;l=183 . If we are batching async requests to the X server and need to wait on them, then flush would help.